photutils/psf/models.py

```diff
@@ -12,8 +12,273 @@
 from ..utils import mask_to_mirrored_num
 from ..extern.nddata_compat import extract_array
 
+__all__ = ['Discrete2DModel', 'IntegratedGaussianPRF', 'PRFAdapter',
+           'prepare_psf_model']
 
-__all__ = ['IntegratedGaussianPRF', 'PRFAdapter', 'prepare_psf_model']
+
+class Discrete2DModel(Fittable2DModel):
+    """
+    A discrete fittable 2D model of an image.
+
+    This class stores a discrete 2D image and computes the values at arbitrary
+    locations (including at intra-pixel, fractional positions) within this
+    image using spline interpolation provided by
+    :py:class:`~scipy.interpolate.RectBivariateSpline`. Even though this
+    particular spline interpolator does not support weighted smoothing,
+    :py:class:`Discrete2DModel` can be used to store image weights
+    so that these can be passed to the fitter.
+
+    The fittable model provided by this class has three model parameters:
+    total flux of the underlying image, and two shifts along each axis of the
+    image.
+
+    Parameters
+    ----------
+    data : numpy.ndarray
+        Array containing 2D image.
+
+    origin : tuple, None, optional
+        Origin of the coordinate system in image pixels. Origin indicates where
+        in the image model coordinates ``x`` and ``y`` are zero. If `origin` is
+        `None`, then model's origin will be set to the center of the image.
+
+    weights : numpy.ndarray, None, optional
+        An array of weights for the corresponding data. [Currently not used]
+
+    fillval : float, optional
+        The value to be returned by the `evaluate` or `__call__` methods
```

Comment:
Doesn't `__call__` or `evaluate` refer to methods within your model?  It's not really `~scipy.interpolate.RectBivariateSpline.__call__` right?